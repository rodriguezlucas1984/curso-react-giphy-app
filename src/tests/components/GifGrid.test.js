import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en  < GifGrind />", () => {
  const category = "Goku";
  let wrapper;
  test("debe de mostrarse correctamente el componente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de mostrarse un titulo h3 con la categoria", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    wrapper = shallow(<GifGrid category={category} />);
    const h3 = wrapper.find("h3");
    expect(h3.text().trim()).toBe(category);
  });
  test("debe de mostart items cuando se cargan imÃ¡genes useFetchGifts", () => {
    const gifs = [
      {
        id: "GRSnxyhJnPsaQy9YLn",
        title: "Dragon Ball GIF by TOEI Animation UK",
        url: "https://media4.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=7e2b37ecd35c5v0pd3ymrio1srwfp17rl6ws43oy7lboy3jq&rid=giphy.gif&ct=g",
      },
      {
        id: "977YesTjNfQC7vQiph",
        title: "Dragon Ball GIF by TOEI Animation UK",
        url: "https://media1.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=7e2b37ecd35c5v0pd3ymrio1srwfp17rl6ws43oy7lboy3jq&rid=giphy.gif&ct=g",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
