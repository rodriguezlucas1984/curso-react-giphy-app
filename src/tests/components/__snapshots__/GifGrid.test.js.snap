// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  < GifGrind /> debe de mostrarse correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
